src/CallbackSerializer.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * This file is part of the prooph/pdo-snapshot-store.
+ * (c) 2016-2017 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2017 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\SnapshotStore;
+
+final class CallbackSerializer implements Serializer
+{
+    /**
+     * callable
+     */
+    private $serializeCallback = 'serialize';
+
+    /**
+     * callable
+     */
+    private $unserializeCallback = 'unserialize';
+
+    public function __construct(?callable $serializeCallback, ?callable $unserializeCallback)
+    {
+        if (null !== $serializeCallback && null !== $unserializeCallback) {
+            $this->serializeCallback = $serializeCallback;
+            $this->unserializeCallback = $unserializeCallback;
+        }
+    }
+
+    /**
+     * @param object|array $data
+     * @return string
+     */
+    public function serialize($data): string
+    {
+        return call_user_func_array($this->serializeCallback, [$data]);
+    }
+
+    /**
+     * @param string $serialized
+     * @return object|array
+     */
+    public function unserialize(string $serialized)
+    {
+        return call_user_func_array($this->unserializeCallback, [$serialized]);
```

Comment:
call_user_func_array is for dynamic length arguments. Use call_user_func as it is faster.

Example: return call_user_func($this->in serializeCallback, $serialized);